crates/orderbook/src/orderbook.rs

```diff
@@ -560,7 +564,24 @@ impl Orderbook {
             OrderEventLabel::Filtered => dto::order::Status::Open,
             OrderEventLabel::Invalid => dto::order::Status::Open,
         };
-        Ok(Some(status))
+        Ok(status)
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
This error is specific to that one endpoint. Defining it on the orderbook as a whole seems wrong.
Might be nicer to define the error types for specific endpoints in that endpoints API handler. Would require that handler to become public in the crate, though. 🤔